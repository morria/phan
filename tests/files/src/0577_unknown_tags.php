<?php

/**
 * @template-something-else Q
 * @template T
 */
class TemplateClass577 {
    /**
     * @param T $v
     */
    public function __construct($v) {
    }

    /**
     * @template-another-thing VV should not warn - this is an unsupported tag that might be used by other tools
     * @template X Phan should warn that it couldn't figure out how X is used
     * @param int $x
     * @param-other-tag VV $x should not warn
     */
    public function test($x) {
        echo strlen($x);
    }
}
