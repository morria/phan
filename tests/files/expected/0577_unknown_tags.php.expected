%s:20 PhanTemplateTypeNotUsedInFunctionReturn Template type X not used in return value of function/method test()
%s:21 PhanTypeMismatchArgumentInternal Argument 1 (string) is int but \strlen() takes string
